src/app/index.ts

```diff
@@ -74,13 +76,32 @@ export default class IlcAppSdk implements IIlcAppSdk {
     }
     /**
      * Isomorphic method to render 404 page.
+     * GLOBAL 404:
      * At SSR in processResponse it sets 404 status code to response.
      * At CSR it triggers global event which ILC listens and renders 404 page.
+     *
+     * CUSTOM 404:
+     * At SSR in processResponse it sets 404 status code and "X-ILC-Override" header to response.
+     * At CSR it renders own not found component from fragment.
      */
-    render404 = () => {
-        if (this.adapter.setStatusCode) {
-            this.adapter.setStatusCode(404);
-        } else {
+    render404: Render404 = (isCustomPage) => {
+        // SSR
+        if (this.adapter.setStatus) {
+            const status: ResponseStatus = {
+                code: 404,
+            };
+
+            if (isCustomPage) {
+                status.headers = {
+                    ['X-ILC-Override']: 'error-page-content',
```

Comment:
I would expect to see this code with HTTP header in SSR code, rather then in AppSdk

Because before AppSdk knew nothing about server communication protocol

Reply:
I would say it's better to work this out through adapters that AppSdk receives at CSR & SSR respectively